tools/integration_tests/run_tests_mounted_directory.sh

```diff
@@ -304,3 +304,122 @@ echo "logging:
 gcsfuse --config-file=/tmp/gcsfuse_config.yaml $TEST_BUCKET_NAME $MOUNT_DIR
 GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/log_rotation/...  -p 1 --integrationTest -v --mountedDirectory=$MOUNT_DIR
 sudo umount $MOUNT_DIR
+
+# Run read cache functional tests.
+function read_cache_test_setup() {
+  cache_size_mb=$1
+  enable_range_read_cache=$2
+  cache_ttl=$3
+
+  cleanup_test_environment
+  generate_config_file "$cache_size_mb" "$enable_range_read_cache" "$cache_ttl"
+}
+
+function cleanup_test_environment() {
+  # Clean up any pre-existing log files.
+  rm -r /tmp/gcsfuse_read_cache_test_logs
+  mkdir /tmp/gcsfuse_read_cache_test_logs
+   # Clean up cache directory.
+  rm -r /tmp/cache-dir
+  mkdir /tmp/cache-dir
+}
+
+function generate_config_file() {
+  cache_size_mb=$1
+  enable_range_read_cache=$2
+  cache_ttl=$3
+
+  echo "logging:
+  file-path: /tmp/gcsfuse_read_cache_test_logs/log.json
+  format: json
+  severity: trace
+file-cache:
+  max-size-mb: $cache_size_mb
+  cache-file-for-range-read: $enable_range_read_cache
+metadata-cache:
+  stat-cache-max-size-mb: 4
+  ttl-secs: $cache_ttl
+  type-cache-max-size-mb: 32
+cache-dir: /tmp/cache-dir" > /tmp/gcsfuse_config.yaml
+}
+
+# cache-file-for-range-read:false read-cache test.
+read_cache_test_setup 50 false 3600
+test_cases=(
+  "TestCacheFileForRangeReadFalseTest/TestRangeReadsWithCacheMiss"
+  "TestCacheFileForRangeReadFalseTest/TestConcurrentReads_ReadIsTreatedNonSequentialAfterFileIsRemovedFromCache"
+)
+for test_case in "${test_cases[@]}"; do
+  gcsfuse --config-file=/tmp/gcsfuse_config.yaml "$TEST_BUCKET_NAME" "$MOUNT_DIR" > /dev/null
+  GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/read_cache/... -p 1 --integrationTest -v --mountedDirectory="$MOUNT_DIR" --testbucket="$TEST_BUCKET_NAME" -run "$test_case"
```

Comment:
I see the read cache test setup_test.go, if we pass mountedDirectory, it only runs the test for mounted Directory. In that case we don't use testbucket?

Reply:
Read cache functional tests uses storage client which requires the bucket name to fetch data from GCS. That is why we are passing both test bucket and mounted directory in these tests.